src/NServiceBus.Core/Transports/HostSettings.cs

```diff
@@ -10,15 +11,23 @@ public class HostSettings
         /// <summary>
         /// Creates a new instance of <see cref="HostSettings"/>.
         /// </summary>
-        public HostSettings(string name, string hostDisplayName, StartupDiagnosticEntries startupDiagnostic, Action<string, Exception> criticalErrorAction, bool setupInfrastructure)
+        public HostSettings(string name, string hostDisplayName, StartupDiagnosticEntries startupDiagnostic, Action<string, Exception> criticalErrorAction,
+            bool setupInfrastructure, SettingsHolder coreSettings)
```

Comment:
```suggestion
        public HostSettings(string name, string hostDisplayName, StartupDiagnosticEntries startupDiagnostic, Action<string, Exception> criticalErrorAction, bool setupInfrastructure, SettingsHolder coreSettings)
```

Reply:
should `coreSettings` be an optional parameter?

Reply:
Makes sense